rclpy/src/rclpy/_rclpy_qos_event.c

```diff
@@ -281,15 +279,19 @@ rclpy_create_event(PyObject * Py_UNUSED(self), PyObject * args)
     return NULL;
   }
 
-  pyevent = PyCapsule_New(event, "rcl_event_t", _destroy_event_capsule);
-  if (!pyevent) {
+  rclpy_handle_t * event_handle = _rclpy_create_handle(event, _destroy_event_capsule);
+  if (!event_handle) {
     ret = rcl_event_fini(event);
     PyMem_Free(event);
     _check_rcl_return(ret, "Failed to fini 'rcl_event_t'");
     return NULL;
   }
-
-  return pyevent;
+  _rclpy_handle_add_dependency(event_handle, parent_handle);
+  if (PyErr_Occurred()) {
+    _rclpy_handle_dec_ref(event_handle);
+  }
+  PyObject * event_capsule = _rclpy_create_handle_capsule(event_handle, "rcl_event_t");
+  return event_capsule;
```

Comment:
@ivanpauno nit: consider collapsing the last two statements.

Reply:
Actually, error handling was wrong there. Fixed in https://github.com/ros2/rclpy/pull/497/commits/c0c56b9b4f24be37b4088154a5afea72821f5651.